src/electionguard_cli/steps/shared/election_record_step.py

```diff
@@ -0,0 +1,82 @@
+from shutil import make_archive
+from os.path import splitext, dirname, basename
+from tempfile import TemporaryDirectory
+from click import echo
+from electionguard import to_file
+from electionguard.constants import get_constants
+
+from electionguard_cli.cli_models import BuildElectionResults, E2eSubmitResults
+from electionguard_cli.cli_models.e2e_decrypt_results import E2eDecryptResults
+from electionguard_cli.cli_models.e2e.e2e_inputs import E2eInputs
+
+from electionguard_tools.helpers.export import export_record
+
+from .cli_step_base import CliStepBase
+
+
+class ElectionRecordStep(CliStepBase):
+    """Responsible for publishing an election record after an election has completed."""
+
+    _COMPRESSION_FORMAT = "zip"
+
+    def export(
+        self,
+        election_inputs: E2eInputs,
+        build_election_results: BuildElectionResults,
+        submit_results: E2eSubmitResults,
+        decrypt_results: E2eDecryptResults,
+    ) -> None:
+
+        self.print_header("Election Record")
+
+        self._export_election_record(
+            election_inputs, build_election_results, submit_results, decrypt_results
+        )
+        self._export_private_keys(election_inputs)
+
+    def _export_election_record(
+        self,
+        election_inputs: E2eInputs,
+        build_election_results: BuildElectionResults,
+        submit_results: E2eSubmitResults,
+        decrypt_results: E2eDecryptResults,
+    ) -> None:
+        guardian_records = [
+            guardian.publish() for guardian in election_inputs.guardians
+        ]
+        constants = get_constants()
+        with TemporaryDirectory() as temp_dir:
+            export_record(
+                election_inputs.manifest,
+                build_election_results.context,
+                constants,
+                [submit_results.device],
+                submit_results.data_store.all(),
+                decrypt_results.plaintext_spoiled_ballots.values(),
+                decrypt_results.ciphertext_tally.publish(),
+                decrypt_results.plaintext_tally,
+                guardian_records,
+                decrypt_results.lagrange_coefficients,
+                election_record_directory=temp_dir,
+            )
+            file_name = splitext(election_inputs.output_record)[0]
+            make_archive(file_name, self._COMPRESSION_FORMAT, temp_dir)
+            echo(f"Exported election record to '{election_inputs.output_record}'")
+
+    def _export_private_keys(self, election_inputs: E2eInputs) -> None:
```

Comment:
This method is new.  If you add `--output-record="./election_record.zip"` to the e2e command this method will export the private keys using the (existing) `guardian.export_private_data()` method so that when you run the `import-ballots` command you can reconstruct the private key data.  Technically the e2e command isn't exactly the right place to do this, it should go into the election setup command that will be built in #592, but it needs to be here for now because there's no other place.